contrib/ci/check-source.py

```diff
@@ -110,6 +112,28 @@ def _test_lines_gerror(self, lines: List[str]) -> None:
                         self.add_failure("uses g_set_error() without using FWUPD_ERROR")
                         break
 
+    def _test_lines_function_length(self, lines: List[str]) -> None:
+        # allowed
+        if os.path.basename(self._current_fn) in [
```

Comment:
I'm not sure it's worth special casing these files, doesn't that basically mean we should have more in common?

Reply:
ohh the comment is bad -- what I meant was "these files are allowed to have a huge main()" -- does this help?

Reply:
Ah yeah

Reply:
I changed it to say that instead :)